apps/server/src/services/rundown-service/rundownCache.ts

```diff
@@ -31,14 +33,27 @@ type NormalisedRundown = Record<EventID, OntimeRundownEntry>;
 let persistedRundown: OntimeRundown = [];
 let persistedCustomFields: CustomFields = {};
 
-/** Utility function gets to expose data */
+/**
+ * Get the cached rundown without triggering regeneration
+ */
 export const getPersistedRundown = (): OntimeRundown => persistedRundown;
 export const getCustomFields = (): CustomFields => persistedCustomFields;
 
-let rundown: NormalisedRundown = {};
+let normalisedRundown: NormalisedRundown = {};
 let order: EventID[] = [];
 let revision = 0;
+
+/**
+ * all mutating functions will set this value if there is a need for re-generation
+ * but will only be cleared by the generate function
+ */
 let isStale = true;
+
+/** Allows safely setting the stale state without accedentaly clearing it */
+const setIsStale = (newValue: boolean) => {
+  isStale = isStale || newValue;
+};
```

Comment:
I didnt really understand what this achieves. Would this not mean that we cannot set isStale to false?

```suggestion
/** Allows safely setting the stale state without accidentaly clearing it */
function setIsStale(newValue: boolean) {
  isStale = isStale || newValue;
};
```